test/unit/server-statefulset.bats

```diff
@@ -894,3 +894,34 @@ load _helpers
       yq '.spec.template.spec.containers[0].command | join(" ") | contains("auto_encrypt = {allow_tls = true}")' | tee /dev/stderr)
   [ "${actual}" = "true" ]
 }
+
+#--------------------------------------------------------------------
+# -bootstrap-expect
+
+@test "server/StatefulSet: -bootstrap-expect defaults to replicas" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -s templates/server-statefulset.yaml  \
+      . | tee /dev/stderr |
+      yq '.spec.template.spec.containers[0].command | join(" ") | contains("-bootstrap-expect=3")' | tee /dev/stderr)
+  [ "${actual}" = "true" ]
+}
+
+@test "server/StatefulSet: -bootstrap-expect can be set by server.bootstrapExpect" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -s templates/server-statefulset.yaml  \
+      --set 'server.bootstrapExpect=5' \
+      . | tee /dev/stderr |
+      yq '.spec.template.spec.containers[0].command | join(" ") | contains("-bootstrap-expect=5")' | tee /dev/stderr)
+  [ "${actual}" = "true" ]
+}
+
+@test "server/StatefulSet: errors if bootstrapExpect < replicas" {
```

Comment:
Is this change backwards compatible? Previously, we would allow `bootstrap expect` to be <= `replicas` but this change would cause such a setting to change.

Reply:
Ooh, you're right! Will add to changelog. I could also remove this but I think this is an invalid configuration.

Reply:
Interesting. I mean, your comment about bootstrapExpect made me realize that my understanding of it was incorrect. I thought it was the min number of servers that needed to be up and running for consensus, which seemed odd but 🤷 . This makes more sense.

Reply:
> To prevent inconsistencies and split-brain (clusters where multiple servers consider themselves leader) situations, you should either specify the same value for -bootstrap-expect or specify no value at all on all the servers. 

From https://www.consul.io/docs/install/bootstrapping